pkg/chartutil/jsonschema.go

```diff
@@ -91,3 +99,51 @@ func ValidateAgainstSingleSchema(values Values, schemaJSON []byte) (reterr error
 
 	return nil
 }
+
+// keep the old behaviour for empty $schema or the one that defined in
+// https://github.com/xeipuuv/gojsonschema/blob/v1.2.0/draft.go#L46-L62
+func validateUsingNewValidator(valuesJSON, schemaJSON []byte) (bool, error) {
+	var partialSchema struct {
+		Schema string `json:"$schema"`
+	}
+	_ = json.Unmarshal(schemaJSON, &partialSchema)
+	if partialSchema.Schema == "" {
+		return false, nil
+	}
+
+	url, err := url.Parse(partialSchema.Schema)
+	if err != nil {
+		return false, nil
+	}
+	if url.Host == "json-schema.org" {
+		switch url.EscapedPath() {
+		case
+			"/draft-04/schema",
+			"/draft-06/schema",
+			"/draft-07/schema":
+			return false, nil
+		}
+	}
+
+	schema, err := newvalidator.UnmarshalJSON(bytes.NewReader(schemaJSON))
+	if err != nil {
+		return true, err
+	}
+	values, err := newvalidator.UnmarshalJSON(bytes.NewReader(valuesJSON))
+	if err != nil {
+		return true, err
+	}
+
+	compiler := newvalidator.NewCompiler()
+	err = compiler.AddResource("file:///values.schema.json", schema)
+	if err != nil {
+		return true, err
+	}
+
+	validator, err := compiler.Compile("file:///values.schema.json")
+	if err != nil {
+		return true, err
+	}
+
+	return true, validator.Validate(values)
```

Comment:
Can you confirm that `validator.Validate` returns human friendly errors please. We need to make sure that any schema validation errors propagate to the user.

Reply:
yes, see the test code

Reply:
nice, thanks